mip/mip.c

```diff
@@ -202,7 +203,11 @@ static bool q_write(struct queue *q, const void *buf, size_t len) {
   return success;
 }
 
-static size_t q_avail(struct queue *q) {
+static inline size_t q_space(struct queue *q) {
+  return q->tail > q->head ? q->tail - q->head : q->tail + (q->len - q->head);
```

Comment:
This is the "modulo-less" way, but is off by 1
I'm wasting one byte to differentiate full from empty as otherwise both pointers point to the same location either on queue empty or queue full. As we won't reach that condition, it won't print q->len when it is actually full, but it always prints available room +1 . I guess we can live with that...